src/main/java/com/kosherjava/zmanim/hebrewcalendar/JewishCalendar.java

```diff
@@ -495,8 +498,11 @@ public int getYomTovIndex() {
 		// check by month (starting from Nissan)
 		switch (getJewishMonth()) {
 		case NISSAN:
+			if (day == 13 && dayOfWeek == 6) {
+					return FAST_OF_THE_FIRSTBORNS;
+			}
 			if (day == 14) {
-				return EREV_PESACH;
+				return EREV_PESACH;//TODO add Fast of firstborns with Erev Pesach
```

Comment:
I ran into a problem here with only being able to return one result. I'm not sure what you want to do with the future of this method, so I just left a TODO comment.

Reply:
does't this  mean that if it's the 13th of the month  and it's friday  that it  will  return that today is taanit  bechorot? don't  we push  up  taanit bechorot to thursday the  12th?

Reply:
@elijulian yes, it will start from Thursday night through Friday. This how the getYomTovIndex method works. It looks at the day of the current date, not the night before it.

Reply:
in essence the 12th is Erev Ta'anit Bechorot

Reply:
@elyahu41 
I would argue that the day (either the 13th or 14th) is never called the "day of _Taanis Bechoros_", rather there is a _minhag_ (brought down _lehalacha_, but still a _minhag_)to fast on _erev pesach_ and that fast that is usually on _Erev Pesach_, but sometimes is on a different date. I would not have the YT index change for this. I was considering adding the day to isTaanis(), but that can honestly be confusing as well since it is not a regular _taanis_. As per the Mishna Berura on 470:2 no. 7, the taanis is only a minhag. In no. 2 the Mishna Berura mentions that it is a _**taanis yachid**_ and that it is preferable to avoid saying _aneinu_ in _chazaras hashatz_ (if there are 10 _bechorim_ fasting we should avoid having a _bechor_ as the shaliach tzibur) since we do not mention a taanis in _Nissan_ publicly. Vayechal is not recited (mentioned in 566:2 in Biur Halacha Veyesh).
In short, my feeling is that this does not fit into the API at this point the way you coded it. Adding an isTaanisBechorim() method is a possibility, but I am not 100% sure about that. The way I would code it is below.

```
/**
 * Return true if the day is Taanis Bechorim (on erev Pesach). It will return true for the 14th of Nissan if it is not
 * on Shabbos, or if the 12th of Nissan occurs on a Thursday
 * 
 * @return true if today is the fast of Bechoros
 */
public boolean isTaanisBechorim() {
    final int day = getJewishDayOfMonth();
    final int dayOfWeek = getDayOfWeek();
    // on 14 Nisan unless that is Shabbos where the fast is moved back to Thursday
    return getJewishMonth() == NISSAN && ((day == 14 && dayOfWeek != Calendar.SATURDAY) ||
            (day == 12 && dayOfWeek == Calendar.THURSDAY ));
}
```

Reply:
@elijulian I double checked, and you were right. I was assuming that the fast was on Friday. My bad. Good thing @KosherJava is here to keep us in check lol.